### PR TITLE
Add watchdog inotify and tests

### PR DESCRIPTION
**Pull request by david415**
This change is [Reviewable](https://reviewable.io/reviews/tahoe-lafs/tahoe-lafs/401)

### CODE REVIEW

**Comment by david415** [comment]
why did that last commit not fix it!?

File: src/allmydata/test/test_inotify.py [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+            raise unittest.SkipTest("twistd does not fork under windows")
+
+    def setUp(self):
+        self.ignore_count = 0
+        self.dirname = filepath.FilePath(self.mktemp())
+        self.dirname.createDirectory()
+        self.inotify = inotify.INotify()
+        self.inotify.startReading()
+        self.addCleanup(self.inotify.stopReading)
+        self.skip_if_windows()
```
Comment (exarkun) [inline src/allmydata/test/test_inotify.py]: Move this up to the beginning of the `setUp` method?

Or do the skip differently:

```
class INotifyTests(unittest.TestCase):
    if runtime.platformType == "win32":
        skip = "twistd does not fork under windows"

    ...
```

I generally prefer setting the skip attribute to raising `SkipTest` during test execution - if it is possible to statically determine the skip that seems to be superior.

Comment (david415) [inline src/allmydata/test/test_inotify.py]: oh duh! thanks

**Comment by exarkun** [comment]
This basically landed in #558

**Pull request closed without merging**